src/main/treeDataProviders/dependenciesTree/dependenciesDataProvider.ts

```diff
@@ -143,24 +181,66 @@ export class DependenciesTreeDataProvider implements vscode.TreeDataProvider<Dep
+                let componentMetadata: IComponentMetadata | undefined = this._treesManager.scanCacheManager.getMetadata(
+                    child.getGoCenterComponentId()
+                );
+                if (componentMetadata) {
+                    // Use xray issue instead of GoCenter
+                    if (!credentialsSet) {
+                        if (componentMetadata.vulnerabilities.severity) {
+                            Translators.severityCountToIssues(componentMetadata.vulnerabilities.severity).forEach(issue => child.issues.add(issue));
+                        }
+                        if (componentMetadata.licenses) {
+                            componentMetadata.licenses.map(Translators.stringToLicense).forEach(license => child.licenses.add(license));
+                            this.filterLicenses.union(child.licenses);
+                        }
+                    }
+                    // Load GoCenter info(e.g. readme url) to each go node
+                    child.componentMetadata = componentMetadata;
+                }
+                this.addGoCenterInfoToTree(child, credentialsSet);
+            }
+        });
+    }
+
+    private async repopulateTree(quickScan: boolean, credentialsSet: boolean) {
         await ScanUtils.scanWithProgress(async (progress: vscode.Progress<{ message?: string; increment?: number }>, checkCanceled: () => void) => {
             this.clearTree();
             let dependenciesTree: DependenciesTreeNode = <DependenciesTreeNode>this.dependenciesTree;
             await DependenciesTreesFactory.createDependenciesTrees(
                 this._workspaceFolders,
                 this._componentsToScan,
+                this._goCenterComponentsToScan,
                 this._treesManager,
                 dependenciesTree,
                 quickScan
             );
-            await this.scanAndCacheArtifact(progress, checkCanceled);
-            for (let dependenciesTreeNode of dependenciesTree.children) {
-                this.addXrayInfoToTree(dependenciesTreeNode);
-                dependenciesTreeNode.issues = dependenciesTreeNode.processTreeIssues();
+            if (credentialsSet) {
+                // Xray + GoCenter users
+                await this.scanAndCacheComponents(progress, checkCanceled);
+                for (let dependenciesTreeNode of dependenciesTree.children) {
+                    this.addXrayInfoToTree(dependenciesTreeNode);
+                    if (dependenciesTreeNode instanceof GoTreeNode) {
+                        this.addGoCenterInfoToTree(dependenciesTreeNode, credentialsSet);
+                    }
+                    dependenciesTreeNode.issues = dependenciesTreeNode.processTreeIssues();
+                }
+            } else {
+                // GoCenter users
+                const totalComponents: number = this._goCenterComponentsToScan.size();
+                vscode.window.showInformationMessage('Xray server is not configured.');
```

Comment:
Please advise with @eyalbe4 about that line because it will always pop for GoCenter users. Maybe we should pop something else, or think about other solution.